atomic_reactor/util.py

```diff
@@ -706,3 +734,25 @@ def get_manifest_digests(image, registry, insecure=False, dockercfg_path=None,
 
     return ManifestDigest(**digests)
 
+
+def get_config_from_registry(image, registry, digest, insecure=False,
+                             dockercfg_path=None, version='v2'):
+    """Return image config by digest
+
+    :param image: ImageName, the remote image to inspect
+    :param registry: str, URI for registry, if URI schema is not provided,
+                          https:// will be used
+    :param insecure: bool, when True registry's cert is not verified
+    :param dockercfg_path: str, dirname of .dockercfg location
+    :param versions: tuple, which manifest schema versions to fetch digest
+
+    :return: dict, versions mapped to their digest
+    """
+    response = query_registry(
+        image, registry, digest=digest, insecure=insecure,
+        dockercfg_path=dockercfg_path, version=version)
+    response_json = response.json()
+    history = response_json['history']
```

Comment:
This is wrong. We aren't looking for response.history[0].v1Compatibility -- we are looking for response, plain and simple, just parsed as JSON.

v1Compatibility is a v2 schema 1 thing. We are not using that for fetching config (and never will).